client/src/Pos/Client/Txp/Util.hs

```diff
@@ -191,6 +192,18 @@ type TxCreateMode m
       , MonadAddresses m
       )
 
+-- | Generic function to create an unsigned transaction, given desired inputs and outputs
+makeUnsignedAbstractTx ::
+                  TxOwnedInputs owner
+               -> TxOutputs
+               -> Either e Tx
+makeUnsignedAbstractTx txInputs outputs = do
+  let
+    tx = UnsafeTx (map snd txInputs) txOutputs txAttributes
+    txOutputs = map toaOut outputs
+    txAttributes = mkAttributes ()
+  pure tx
```

Comment:
I mentioned this in the other PR, but there's no reason for this to be an `Either`. Just have it return the `Tx` directly.